tests/conditional_formatting/conditional_formatting_plugin.test.ts

```diff
@@ -679,6 +680,199 @@ describe("conditional format", () => {
+      ["=A5", "=F5"],
+      ["=$F$5", "=$F$10"],
+      ["=F5*5", "=F5*10"],
+      ["=SUM(A1:F5)", "=SUM(A1:F5)"],
+    ];
+
+    function cellIsRuleEqual(f: string, useless: string): ConditionalFormatRule {
+      return {
+        type: "CellIsRule",
+        values: [f],
+        operator: "Equal",
+        style: { fillColor: "orange" },
+      };
+    }
+
+    function cellIsRuleBetween(f: string, f2: string): ConditionalFormatRule {
+      return {
+        type: "CellIsRule",
+        values: [f, f2],
+        operator: "Between",
+        style: { fillColor: "orange" },
+      };
+    }
+
+    function colorScaleRule(min: string, max: string): ConditionalFormatRule {
+      return {
+        type: "ColorScaleRule",
+        minimum: {
+          type: "formula",
+          value: min,
+          color: 0,
+        },
+        maximum: {
+          type: "formula",
+          value: max,
+          color: 0,
+        },
+      };
+    }
+
+    function IconSetRule(up: string, lw: string): ConditionalFormatRule {
+      return {
+        type: "IconSetRule",
+        icons: {
+          upper: "u",
+          middle: "m",
+          lower: "l",
+        },
+        upperInflectionPoint: {
+          type: "formula",
+          operator: "gt",
+          value: up,
+        },
+        lowerInflectionPoint: {
+          type: "formula",
+          operator: "gt",
+          value: lw,
+        },
+      };
+    }
+
+    function ruleToCF(rule: ConditionalFormatRule): ConditionalFormat {
+      return {
+        id: "1",
+        rule,
+        ranges: ["A1"],
+      };
+    }
+
+    function formulasToCFs(formulas: string[][], cfrule: any): ConditionalFormat[] {
+      return formulas.map((fn) => cfrule(fn[0], fn[1])).map(ruleToCF);
+    }
+
+    const CFs = [
+      ["cellIsRuleEqual", cellIsRuleEqual],
+      ["cellIsRuleBetween", cellIsRuleBetween],
+      ["colorScaleRule", colorScaleRule],
+      ["IconSetRule", IconSetRule],
+    ];
+
+    test.each(CFs)("On row deletion %s", (name, cfrule) => {
+      const expected = formulasToCFs(
+        [
+          ["=A1", "=A3"],
+          ["=F1", "=F3"],
+          ["=A3", "=F3"],
+          ["=$F$3", "=$F$8"],
+          ["=F3*5", "=F3*10"],
+          ["=SUM(A1:F3)", "=SUM(A1:F3)"],
+        ],
+        cfrule
+      );
+
+      model = new Model({
+        sheets: [
+          {
+            colNumber: 7,
+            rowNumber: 4,
+            cells: { C3: "42" },
+            conditionalFormats: formulasToCFs(formulas, cfrule),
+          },
+        ],
+      });
+      deleteRows(model, [1, 3]);
+      expect(model.getters.getConditionalFormats(sheetId)).toEqual(expected);
+    });
+
+    test.each(CFs)("On column deletion %s", (name, cfrule) => {
+      const expected = formulasToCFs(
+        [
+          ["=A1", "=A5"],
+          ["=D1", "=D5"],
+          ["=A5", "=D5"],
+          ["=$D$5", "=$D$10"],
+          ["=D5*5", "=D5*10"],
+          ["=SUM(A1:D5)", "=SUM(A1:D5)"],
```

Comment:
The tests are kinda hard to read with the original formulas 100+ lines above the expected result. Maybe we could chanhe the structure with the test.each having a list of "(formula, expectedResult) " and the test body looping on every CF type. What do you think ?